libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -91,11 +109,26 @@ func New(cfg *common.Config) (processors.Processor, error) {
 		return nil, fmt.Errorf("Can not initialize kubernetes plugin with zero matcher plugins")
 	}
 
+	processor := &kubernetesAnnotator{
+		indexers: indexers,
+		matchers: matchers,
+		cache:    newCache(config.CleanupTimeout),
+	}
+
 	client, err := kubernetes.GetKubernetesClient(config.KubeConfig)
 	if err != nil {
-		return nil, err
+		logp.Debug("kubernetes", "%v: could not create kubernetes client with config: %v", "add_kubernetes_metadata", config.KubeConfig)
```

Comment:
Are we silently failing here? probably this should be an info if kubernetes metadata wont be available. But then, we are returning the processor. Probably that makes sense, but I'm not familiar with the code yet.

In any case, should we consider an `Info` instead of a `Debug` here? It is probably not needed, but a nice extra to log `InCluster` instead of the empty `config.KubeConfig` when the later is empty.

Reply:
Yeap kind of "silent_failure" with the proper logging is what we can do in this autodetection approach.